src/index.ts

```diff
@@ -73,37 +92,33 @@ export const compare = memoize((versionA: VersionInput, versionB: VersionInput):
 	versionA = normalize(versionA);
 	versionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
-	if (isAValid && !isBValid) {
-		return 1;
-	}
-	if (!isAValid && isBValid) {
-		return -1;
-	}
-	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+	const semverA = semver.parse(versionA);
```

Comment:
`semver.valid`, `semver.compare` both do a full parse under the hood, so hopefully this is a bit faster.